File: ZaeXT/internal/handler/response/sse.go
package response

type OpenAIStreamChoiceDelta struct {
	Content string `json:"content"`
}

type OpenAIStreamChoice struct {
	Delta OpenAIStreamChoiceDelta `json:"delta"`
}

type OpenAIStreamResponse struct {
	Choices []OpenAIStreamChoice `json:"choices"`
}

File: ZaeXT/frontend/src/types/sse.ts
import type { MessageRole } from './conversation'

export interface OpenAIStreamChoiceDelta {
  role?: MessageRole
  content?: string
  function_call?: {
    name?: string
    arguments?: string
  }
}

export interface OpenAIStreamChoice {
  index?: number
  delta: OpenAIStreamChoiceDelta
  finish_reason?: string | null
}

export interface OpenAIStreamChunk {
  id?: string
  object?: string
  created?: number
  model?: string
  choices: OpenAIStreamChoice[]
}

export type SSEMessageHandler = (fragment: string, chunk: OpenAIStreamChunk) => void

export interface StreamCallbacks {
  onDelta: SSEMessageHandler
  onDone: () => void
  onError?: (error: Error) => void
}

export interface StreamController {
  abort: () => void
}
